### PR TITLE
Raw Image support

### PR DESCRIPTION
**Pull request by detiber**
What this PR does / why we need it:

Rebased and updated version of #496 

Which issue(s) this PR fixes (optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged): Fixes #496

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
There was an Azure AD incident but it should be recovered now https://status.azure.com/en-us/status

/retest

**Comment by CecileRobertMichon** [comment]
it looks like the windows builds are failing, @jsturtevant can you please take a look?

```
[0;32m    vhd-windows-2004: fatal: [default]: FAILED! => {"changed": true, "cmd": "# https://github.com/microsoft/azure-pipelines-image-generation/pull/1259/files\n$service = Get-Service \"Windefend\"\n$service.WaitForStatus(\"Running\",\"00:5:00\")\nUpdate-MpSignature", "delta": "0:00:14.397761", "end": "2021-03-16 02:34:54.638147", "msg": "non-zero return code", "rc": 1, "start": "2021-03-16 02:34:40.240385", "stderr": "Update-MpSignature : The remote procedure call failed. \r\nAt line:4 char:1\r\n+ Update-MpSignature\r\n+ ~~~~~~~~~~~~~~~~~~\r\n    + CategoryInfo          : NotSpecified: (MSFT_MpSignature:ROOT\\Microsoft\\...SFT_MpSignature) [Update-MpSignature], \r\n    CimException\r\n    + FullyQualifiedErrorId : HRESULT 0x800706be,Update-MpSignature", "stderr_lines": ["Update-MpSignature : The remote procedure call failed. ", "At line:4 char:1", "+ Update-MpSignature", "+ ~~~~~~~~~~~~~~~~~~", "    + CategoryInfo          : NotSpecified: (MSFT_MpSignature:ROOT\\Microsoft\\...SFT_MpSignature) [Update-MpSignature], ", "    CimException", "    + FullyQualifiedErrorId : HRESULT 0x800706be,Update-MpSignature"], "stdout": "", "stdout_lines": []}[0m
[0;32m    vhd-windows-2004:[0m
[0;32m    vhd-windows-2004: PLAY RECAP *********************************************************************[0m
[0;32m    vhd-windows-2004: default                    : ok=5    changed=2    unreachable=0    failed=1    skipped=2    rescued=0    ignored=0[0m
[0;32m    vhd-windows-2004:[0m
[1;32m==> vhd-windows-2004: Provisioning step had errors: Running the cleanup provisioner, if present...[0m
[1;32m==> vhd-windows-2004: Removing the created Deployment object: 'pkrdpf5g0bzehll'[0m
[0;32m    vhd-windows-2019: fatal: [default]: FAILED! => {"changed": true, "cmd": "# https://github.com/microsoft/azure-pipelines-image-generation/pull/1259/files\n$service = Get-Service \"Windefend\"\n$service.WaitForStatus(\"Running\",\"00:5:00\")\nUpdate-MpSignature", "delta": "0:00:18.733891", "end": "2021-03-16 02:35:01.387296", "msg": "non-zero return code", "rc": 1, "start": "2021-03-16 02:34:42.653405", "stderr": "Update-MpSignature : The remote procedure call failed. \r\nAt line:4 char:1\r\n+ Update-MpSignature\r\n+ ~~~~~~~~~~~~~~~~~~\r\n    + CategoryInfo          : NotSpecified: (MSFT_MpSignature:ROOT\\Microsoft\\...SFT_MpSignature) [Update-MpSignature], \r\n    CimException\r\n    + FullyQualifiedErrorId : HRESULT 0x800706be,Update-MpSignature", "stderr_lines": ["Update-MpSignature : The remote procedure call failed. ", "At line:4 char:1", "+ Update-MpSignature", "+ ~~~~~~~~~~~~~~~~~~", "    + CategoryInfo          : NotSpecified: (MSFT_MpSignature:ROOT\\Microsoft\\...SFT_MpSignature) [Update-MpSignature], ", "    CimException", "    + FullyQualifiedErrorId : HRESULT 0x800706be,Update-MpSignature"], "stdout": "", "stdout_lines": []}[0m
[0;32m    vhd-windows-2019:[0m
[0;32m    vhd-windows-2019: PLAY RECAP *********************************************************************[0m
[0;32m    vhd-windows-2019: default                    : ok=5    changed=2    unreachable=0    failed=1    skipped=2    rescued=0    ignored=0[0m
[0;32m    vhd-windows-2019:[0m
[1;32m==> vhd-windows-2019: Provisioning step had errors: Running the cleanup provisioner, if present...[0m
[1;32m==> vhd-windows-2019: Removing the created Deployment object: 'pkrdpm24mw4u3hj'[0m
[1;32m==> vhd-windows-2004: Removing the created Deployment object: 'kvpkrdpf5g0bzehll'[0m
[1;32m==> vhd-windows-2019: Removing the created Deployment object: 'kvpkrdpm24mw4u3hj'[0m
[1;32m==> vhd-windows-2004: 
==> vhd-windows-2004: Cleanup requested, deleting resource group ...[0m
[1;32m==> vhd-windows-2019: 
==> vhd-windows-2019: Cleanup requested, deleting resource group ...[0m
```

**Comment by CecileRobertMichon** [comment]
https://github.com/kubernetes-sigs/image-builder/issues/549

**Comment by detiber** [comment]
@CecileRobertMichon thanks for the assist on the test failure
@codenrhoden could use a review if you have some time

File: images/capi/packer/raw/packer.json [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    "build_timestamp": "{{timestamp}}",
+    "containerd_sha256": null,
+    "containerd_url": "https://github.com/containerd/containerd/releases/download/v{{user `containerd_version`}}/cri-containerd-cni-{{user `containerd_version`}}-linux-amd64.tar.gz",
+    "containerd_version": null,
+    "cpus": "1",
+    "crictl_url": "https://github.com/kubernetes-sigs/cri-tools/releases/download/v{{user `crictl_version`}}/crictl-v{{user `crictl_version`}}-linux-amd64.tar.gz",
+    "crictl_version": null,
+    "disk_size": "6144",
+    "existing_ansible_ssh_args": "{{env `ANSIBLE_SSH_ARGS`}}",
+    "format": "raw",
```
Comment (codenrhoden) [inline images/capi/packer/raw/packer.json:114]: I'm sure I'm missing important details, but at first glance, the only difference between this new "raw" builder and the existing "qemu" builder is this format option -- "raw" vs "qcow2". It's configured with the same builder type, same accelerator, same pre-req's.

So my naive question is what prevents "raw" from being a configurable option in the existing qemu builder vs adding a whole new one?

Comment (detiber) [reply images/capi/packer/raw/packer.json:114]: @codenrhoden The main difference is the target environment, with the raw images we are targeting physical hardware vs virtual. While we probably could make all of the differences configurable, I think the separate targets help avoid confusion and avoid introducing potential issues to the other target inadvertently.

Also, I expect things will continue to diverge more as we continue testing and optimizing the raw images for use with bare metal provisioning tooling, such as https://tinkerbell.org/

**Comment by codenrhoden** [comment]
/lgtm

Thanks @detiber !

**Comment by thebsdbox** [comment]
yay !

**Pull request merged**